src/org/labkey/targetedms/TargetedMSManager.java

```diff
@@ -520,6 +521,31 @@ public static TableInfo getTableInfoListItemValue() {
         return getSchema().getTable(TargetedMSSchema.TABLE_LIST_ITEM_VALUE);
     }
 
+    public static TableInfo getTableInfoBibliospec()
+    {
+        return getSchema().getTable(TargetedMSSchema.TABLE_BIBLIOSPEC_LIB_INFO);
+    }
+
+    public static TableInfo getTableInfoHunterLib()
+    {
+        return getSchema().getTable(TargetedMSSchema.TABLE_HUNTER_LIB_INFO);
+    }
+
+    public static TableInfo getTableInfoNistLib()
+    {
+        return getSchema().getTable(TargetedMSSchema.TABLE_NIST_LIB_INFO);
+    }
+
+    public static TableInfo getTableInfoSpectrastLib()
+    {
+        return getSchema().getTable(TargetedMSSchema.TABLE_SPECTRAST_LIB_INFO);
+    }
+
+    public static TableInfo getTableInfoChromatogramLib()
+    {
+        return getSchema().getTable(TargetedMSSchema.TABLE_CHROMATOGRAM_LIB_INFO);
```

Comment:
We need to handle these tables in the createTable() method too. They don't have a Container column, so we need to join via the PrecursorId column to filter the rows to show just the ones in the current container. It looks like the implementation for PrecursorLibInfo doesn't do this, which is a security concern as we expose this table's data from ALL folders in ALL folders.

I think this can be done easily by returning a TargetedMSTable that uses ContainerJoinType.PrecursorFK

Reply:
This comment applies to TargetedMSSchema, not TargetedMSManager